File: 2024-2025/repetitions1.cs
using System;

namespace Repetitionsopgave
{
    class Program
    {
        static void Main(string[] args)
        {
            while (true)
            {
                Console.WriteLine("Menu:");
                Console.WriteLine("1) Vurdering af tal");
                Console.WriteLine("2) Udskriv liste af lige tal");
                Console.WriteLine("3) Alder");
                Console.WriteLine("4) Afslut");
                Console.Write("Vælg en mulighed: ");
                string choice = Console.ReadLine();

                switch (choice)
                {
                    case "1":
                        EvaluateNumbers();
                        break;
                    case "2":
                        PrintEvenNumbers();
                        break;
                    case "3":
                        CalculateAge();
                        break;
                    case "4":
                        return;
                    default:
                        Console.WriteLine("Ugyldigt valg. Prøv igen.");
                        break;
                }
            }
        }

        static void EvaluateNumbers()
        {
            Console.WriteLine("Indtast 4 heltal:");
            int[] numbers = new int[4];
            for (int i = 0; i < 4; i++)
            {
                Console.Write($"Tal {i + 1}: ");
                numbers[i] = int.Parse(Console.ReadLine());
            }

            int evenCount = 0;
            int oddCount = 0;
            int greaterThan100Count = 0;
            int negativeCount = 0;

            foreach (int num in numbers)
            {
                if (num % 2 == 0)
                    evenCount++;
                else
                    oddCount++;

                if (num > 100)
                    greaterThan100Count++;

                if (num < 0)
                    negativeCount++;
            }

            Console.WriteLine($"Antal lige tal: {evenCount}");
            Console.WriteLine($"Antal ulige tal: {oddCount}");
            Console.WriteLine($"Antal tal større end 100: {greaterThan100Count}");
            Console.WriteLine($"Antal negative tal: {negativeCount}");
        }

        static void PrintEvenNumbers()
        {
            Console.WriteLine("Indtast to heltal:");
            Console.Write("Første tal: ");
            int num1 = int.Parse(Console.ReadLine());
            Console.Write("Andet tal: ");
            int num2 = int.Parse(Console.ReadLine());

            int min = Math.Min(num1, num2);
            int max = Math.Max(num1, num2);

            Console.WriteLine("Lige tal mellem de to tal:");
            for (int i = min; i <= max; i++)
            {
                if (i % 2 == 0)
                    Console.WriteLine(i);
            }

            int divisionCount = max / min;
            Console.WriteLine($"Det mindste tal går op i det største {divisionCount} gange.");
        }

        static void CalculateAge()
        {
            Console.WriteLine("Indtast din fødselsdato:");
            Console.Write("År: ");
            int year = int.Parse(Console.ReadLine());
            Console.Write("Måned: ");
            int month = int.Parse(Console.ReadLine());
            Console.Write("Dag: ");
            int day = int.Parse(Console.ReadLine());

            DateTime birthDate = new DateTime(year, month, day);
            DateTime currentDate = DateTime.Now;

            int age = currentDate.Year - birthDate.Year;
            if (currentDate < birthDate.AddYears(age))
                age--;

            Console.WriteLine($"Din alder er: {age} år.");
        }
    }
}

File: 2024/repetitions1.py
import datetime

def main():
    while True:
        print("Menu:")
        print("1) Vurdering af tal")
        print("2) Udskriv liste af lige tal")
        print("3) Alder")
        print("4) Afslut")
        choice = input("Vælg en mulighed: ")

        if choice == "1":
            evaluate_numbers()
        elif choice == "2":
            print_even_numbers()
        elif choice == "3":
            calculate_age()
        elif choice == "4":
            break
        else:
            print("Ugyldigt valg. Prøv igen.")

def evaluate_numbers():
    print("Indtast 4 heltal:")
    numbers = []
    for i in range(4):
        num = int(input(f"Tal {i + 1}: "))
        numbers.append(num)

    even_count = sum(1 for num in numbers if num % 2 == 0)
    odd_count = sum(1 for num in numbers if num % 2 != 0)
    greater_than_100_count = sum(1 for num in numbers if num > 100)
    negative_count = sum(1 for num in numbers if num < 0)

    print(f"Antal lige tal: {even_count}")
    print(f"Antal ulige tal: {odd_count}")
    print(f"Antal tal større end 100: {greater_than_100_count}")
    print(f"Antal negative tal: {negative_count}")

def print_even_numbers():
    print("Indtast to heltal:")
    num1 = int(input("Første tal: "))
    num2 = int(input("Andet tal: "))

    min_num = min(num1, num2)
    max_num = max(num1, num2)

    print("Lige tal mellem de to tal:")
    for i in range(min_num, max_num + 1):
        if i % 2 == 0:
            print(i)

    division_count = max_num // min_num
    print(f"Det mindste tal går op i det største {division_count} gange.")

def calculate_age():
    print("Indtast din fødselsdato:")
    year = int(input("År: "))
    month = int(input("Måned: "))
    day = int(input("Dag: "))

    birth_date = datetime.date(year, month, day)
    current_date = datetime.date.today()

    age = current_date.year - birth_date.year
    if current_date < birth_date.replace(year=current_date.year):
        age -= 1

    print(f"Din alder er: {age} år.")

if __name__ == "__main__":
    main()
